src/writer.rs

```diff
@@ -131,13 +131,14 @@ impl<W> Writer<W> {
     ///     // We need to provide error type, because it is not named somewhere explicitly
     ///     .write_inner_content::<_, Error>(|writer| {
     ///         let fruits = ["apple", "orange"];
+    ///         let mut n = 0;
     ///         for (quant, item) in fruits.iter().enumerate() {
-    ///             writer
+    ///             n += writer
     ///                 .create_element("fruit")
     ///                 .with_attribute(("quantity", quant.to_string().as_str()))
     ///                 .write_text_content(BytesText::new(item))?;
     ///         }
-    ///         Ok(())
+    ///         Ok(n)
```

Comment:
Honestly, I think I would prefer not following this pattern for the `ElementWriter` APIs.  You're right, it's messy.

The `ElementWriter` APIs are just for convenience, you can't do anything that you can't also do with the lower level ones.  So I think as long as it's possible to get the behavior you want, we can keep these APIs as convenient and simple as possible.

Reply:
Alright, I'll undo those changes and try to finalize this weekend.